### PR TITLE
Only write changed values for authui v2 customization

### PR DESCRIPTION
**Pull request by elise-ng**
Ref DEV-1602
Based on #4486 

https://github.com/user-attachments/assets/245b8822-2866-4b96-9178-ed4dcbe7ebe7

### CODE REVIEW

**Comment by elise-ng** [comment]
@louischan-oursky

**Comment by elise-ng** [comment]
@fungc-io plz see if the ux is ok to you?

**Review by louischan-oursky — COMMENTED** [review]
1. The style is not saved to the correct style sheet. For example, when Theme is Auto, the button border is saved to `authgear-authflowv2-light-theme.css` with the `:root:not(.dark)` selector. So it has no effect in the dark mode.
2. When I revert all my changes, an empty `:root.dark {}` or `:root:not(.dark)` is saved. In this case, we should remove the file.
3. The toggle has no selected state is not a good UX. I have no idea what the default state is. Instead, we should remove the style if the style being saved is the default.

For Point 1, I think we should

1. If the Theme is Light, then we always remove the dark theme file.
2. If the Theme is Dark, then we always remove the light theme file.
4. If the Theme is Auto, then we always write the light theme file and the dark theme file with identical contents, only the selector is different.

**Comment by fungc-io** [comment]
> @fungc-io plz see if the ux is ok to you?

@elise-ng ,
the `0` that appears when you selected the border radius style is weird. it'd be great if it shows a placeholder with a value and unit
otherwise is cool

**Comment by elise-ng** [comment]
@louischan-oursky 

> when Theme is Auto, the button border is saved to authgear-authflowv2-light-theme.css with the :root:not(.dark) selector. So it has no effect in the dark mode.

seems i missed some design details for auto mode, there will be options for both modes for each field and a switch for the preview

**Comment by elise-ng** [comment]
> The toggle has no selected state is not a good UX. I have no idea what the default state is. Instead, we should remove the style if the style being saved is the default.

discussed with fung, we will show the default as active, but save css entry if user click another option and back to the default one

**Comment by elise-ng** [comment]
also, should change wording for theme option to "light mode only", "dark mode only" and "auto based on user preference"

File: portal/src/components/design/BorderRadius.tsx [hunk 1]
```diff
@@ -63,14 +63,21 @@ const BorderRadius: React.VFC<BorderRadiusProps> = function BorderRadius(
...
+    if (value == null) {
+      return;
+    }
+    setRadiusValue(value);
+  }, []);
+
+  const onBorderRadiusBlur = useCallback(
+    (ev: React.FocusEvent<HTMLInputElement>) => {
+      if (ev.target.value === "") {
+        onChange(undefined);
```
[the remark this answers is not in the document]
Comment (elise-ng) [reply portal/src/components/design/BorderRadius.tsx]: yes, fixed 🙏

**Comment by louischan-oursky** [comment]
Please ping me again when this is ready for review.

**Comment by elise-ng** [comment]
@louischan-oursky updated:


https://github.com/user-attachments/assets/6b69fd95-838b-4126-b593-9fcfdd98e16f

**Comment by louischan-oursky** [comment]
typecheck failed.

**Comment by louischan-oursky** [comment]
1. Not sure if it is my problem. The preview does not work at all. I see no error messages in the console.

https://github.com/user-attachments/assets/00f9d9f5-1c89-4db7-bce3-537d1e8f9e2b

2. The color picker has a warning about uncontrolled component. Please fix it.

**Comment by louischan-oursky** [comment]
Added a commit to fix Point 1. But the non-theme aware property (like card-alignment) still not work in preview. The color picker warning is still there. I tried to fix it but it does not seem to be easy.

I think we should improve the color picker
1. When I am typing, and the color is partial (like "yello"), the small square color indicator should show a static gray color to indicator the color is incomplete
2. After fixing the uncontrolled component, the text input should work as before. Try testing this case, input "green", see green color in the small square indicator, press backspace once, the input should become "gree", the indicator is in gray, see no warning in the console.

**Comment by elise-ng** [comment]
@louischan-oursky 

> The color picker has a warning about uncontrolled component. Please fix it.

updated

> When I am typing, and the color is partial (like "yello"), the small square color indicator should show a static gray color to indicator the color is incomplete

after the fix the preview will show the default themed color for any invalid input string:

https://github.com/user-attachments/assets/446afd9e-a9df-49b0-a8ab-a96f72402e7a

(@fungc-io plz confirm if this ui/ux is ok?)

**Comment by elise-ng** [comment]
@louischan-oursky seems e2e test is failing from master?

**Comment by louischan-oursky** [comment]
@elise-ng As I said

> But the non-theme aware property (like card-alignment) still not work in preview.


https://github.com/user-attachments/assets/083c2958-ffa0-4e16-8ef3-b2b92304ad3d

**Comment by louischan-oursky** [comment]
> seems e2e test is failing from master?

Yes, it was. It was fixed in https://github.com/authgear/authgear-server/pull/4499 You can rebase.

**Comment by louischan-oursky** [comment]
In dark theme, removing background color does not revert to default color

https://github.com/user-attachments/assets/991b0d64-d769-42d6-97cc-eeebc93696c6

I guess there are other cases that the preview does not work. Can you test it more thoroughly?

**Comment by fungc-io** [comment]
@elise-ng ,
> after the fix the preview will show the default themed color for any invalid input string:
> plz confirm if this ui/ux is ok?

Yes, that's great

**Comment by elise-ng** [comment]
@louischan-oursky 

> I guess there are other cases that the preview does not work. Can you test it more thoroughly?

checked each config and fixed issues found:

https://github.com/user-attachments/assets/05efe28e-e027-487b-9fe3-9c0a7db57861

(this is from the last pr on the chain)

**Comment by louischan-oursky** [comment]
Thank you.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**